### PR TITLE
Purge 'mapping_2d::'.

### PR DESCRIPTION
**Pull request by pifon2a**
[Code structure RFC](https://github.com/pifon2a/rfcs/blob/e11bca586f8cc733063ea9c94bdade71086ed9ea/text/0000-code-structure.md)

### CODE REVIEW

File: cartographer/mapping_3d/local_trajectory_builder_options_3d.cc [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #include "cartographer/mapping_3d/local_trajectory_builder_options_3d.h"
 
 #include "cartographer/internal/mapping/motion_filter.h"
-#include "cartographer/mapping_2d/scan_matching/real_time_correlative_scan_matcher.h"
+#include "cartographer/mapping_2d/scan_matching/real_time_correlative_scan_matcher_2d.h"
```
Comment (gaschler) [inline cartographer/mapping_3d/local_trajectory_builder_options_3d.cc]: mapping_3d includes 2d?

File: cartographer/mapping_2d/scan_matching/translation_delta_cost_functor_2d.h [hunk 2]
```diff
@@ -44,23 +45,23 @@ class TranslationDeltaCostFunctor {
  private:
   // Constructs a new TranslationDeltaCostFunctor from the given
```
Comment (gaschler) [inline cartographer/mapping_2d/scan_matching/translation_delta_cost_functor_2d.h]: TranslationDeltaCostFunctor2D

File: cartographer/mapping_2d/pose_graph/spa_cost_function_2d.h [hunk 3]
```diff
@@ -30,25 +30,23 @@
...
 
 namespace cartographer {
-namespace mapping_2d {
+namespace mapping {
 namespace pose_graph {
 
-class SpaCostFunction {
+class SpaCostFunction2D {
  public:
-  using Constraint = mapping::PoseGraph::Constraint;
```
Comment (gaschler) [inline cartographer/mapping_2d/pose_graph/spa_cost_function_2d.h]: unrelated change?

File: cartographer/mapping/trajectory_builder_interface.h [hunk 4]
```diff
@@ -107,7 +107,7 @@ class TrajectoryBuilderInterface {
                              const sensor::LandmarkData& landmark_data) = 0;
   // Allows to directly add local SLAM results to the 'PoseGraph'. Note that it
   // is invalid to add local SLAM results for a trajectory that has a
-  // 'LocalTrajectoryBuilder'.
+  // 'LocalTrajectoryBuilder2D'.
```
Comment (gaschler) [inline cartographer/mapping/trajectory_builder_interface.h]: 2D or 3D

File: cartographer/mapping_2d/pose_graph/spa_cost_function_2d.h (hunk 3 above)
Comment (pifon2a) [reply cartographer/mapping_2d/pose_graph/spa_cost_function_2d.h]: After removing 'mapping::' from this typedef, there seems to be no reason to keep it at all. Inlined it.

File: cartographer/mapping_3d/local_trajectory_builder_options_3d.cc (hunk 1 above)
Comment (pifon2a) [reply cartographer/mapping_3d/local_trajectory_builder_options_3d.cc]: Done, moved to a separate file.

File: cartographer/internal/mapping/global_trajectory_builder.cc [hunk 5]
```diff
@@ -48,7 +48,7 @@ class GlobalTrajectoryBuilder : public mapping::TrajectoryBuilderInterface {
       const std::string& sensor_id,
       const sensor::TimedPointCloudData& timed_point_cloud_data) override {
     CHECK(local_trajectory_builder_)
-        << "Cannot add TimedPointCloudData without a LocalTrajectoryBuilder.";
+        << "Cannot add TimedPointCloudData without a LocalTrajectoryBuilder2D.";
```
[the remark this answers is not in the document]
Comment (pifon2a) [reply cartographer/internal/mapping/global_trajectory_builder.cc]: Done

File: cartographer/mapping/trajectory_builder_interface.h (hunk 4 above)
Comment (pifon2a) [reply cartographer/mapping/trajectory_builder_interface.h]: Done
